sljit_src/sljitNativeRISCV_common.c

```diff
@@ -1669,6 +1763,20 @@ static SLJIT_INLINE sljit_s32 emit_single_op(struct sljit_compiler *compiler, sl
 
 	case SLJIT_ROTL:
 	case SLJIT_ROTR:
+#if defined __riscv_zbb  
+		if (flags & SRC2_IMM) {
+			SLJIT_ASSERT(src2 != 0);
```

Comment:
wouldn't it be better to return SLJIT_SUCCESS for this?, also no need to duplicate logic IMHO

Reply:
How would you merge the code?

Reply:
I meant that there is no need to have it twice if it is outside the `#ifdef`

Reply:
A few lines could be shared, but more don't. This is why I ask how could this look better?